jquery.flexslider.js

```diff
@@ -568,15 +571,34 @@
           }
           else if (vertical) { //VERTICAL:
             slider.viewport.height(slider.h);
-            slider.setProps(slider.h, "setTotal");
+            slider.setProps(slider.h, "setCurrent");
           } else {
+            slider.newSlides.width(slider.computedW);
+            // VARIABLE IMAGE WIDTH:
+            if (variableImageWidth) methods.variableImageWidth();
             // SMOOTH HEIGHT:
             if (slider.vars.smoothHeight) methods.smoothHeight();
-            slider.newSlides.width(slider.computedW);
-            slider.setProps(slider.computedW, "setTotal");
+            slider.setProps(slider.computedW, "setCurrent");
           }
         }
       },
+      variableImageWidth: function() {
+        var minHeight = 0;
+        slider.newSlides.each(function(index) {
+          $(this).children('img').first().height('auto');
```

Comment:
Why only use `img` elements to determine slide height? What if a slide has no `img` element?